### PR TITLE
feat: add exclude regex array for generated pages

### PR DESCRIPTION
**Pull request by TheAlexLichter**
This PR will add a property called `generate.exclude`. It accepts an array of regular expressions and will prevent generation of routes matching them. The routes will still be accessible when `generate.fallback` is used.

## Use cases

* Having routes like `/admin` in the Nuxt app that don't need to be generated upfront but still need to be accessible
* Modules like [sitemap-module](https://github.com/nuxt-community/sitemap-module/blob/dev/src/index.js#L17) could use it as default for excluding routes in `generate` mode.

## Usage

```js
// nuxt.config.js
export default {
  // ...
  generate: {
    //...
    exclude: [/^\/admin/] // exclude every URL starting with "/admin"
  }
}
```

## Testing

Testing this feature is a bit tricky because we never test that the `generated` fallback actually loads the routes in SPA mode. We only test that it has been generated correctly [here](https://github.com/nuxt/nuxt.js/blob/fe0516978a650166bf178c0339dd7657c29e2504/test/unit/fallback.generate.test.js).

## Types of changes
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [x] All new and existing tests are passing.

### CODE REVIEW

**Comment by manniL** [comment]
image

File: packages/generator/src/generator.js [hunk 1]
```diff
@@ -83,6 +83,11 @@ export default class Generator {
       this.options.router.mode === 'hash'
         ? ['/']
         : flatRoutes(this.options.router.routes)
+
+    this.options.generate.exclude.forEach((regex) => {
+      routes = routes.filter(route => !regex.test(route))
```
Comment (pi0) [inline packages/generator/src/generator.js]: each .filter` creates a clone of the array. So can we use something like `filter(route =>  this.options.generate.exclud.some()` ?

**Review by pi0 — APPROVED** [review]
LGTM. BTW do you mind adding a small test for it? @manniL

**Comment by manniL** [comment]
@pi0 That might be a *bit* problematic as I can only test that the file is *not* generated. I can't test if it can be correctly accessed through the `fallback` file. (see "Testing" in the PR text).

Would that suffice?

**Comment by pi0** [comment]
@manniL Seems OK 👍😊

**Pull request merged**